### PR TITLE
refactor: Improve spokepool script error handling

### PR DESCRIPTION
**Pull request by pxrl**
Follow the node.js guidance by setting process.exitCode rather than exiting directly. This aligns with what was already implemented in the HubPool script.

### CODE REVIEW

**Comment by pxrl** [comment]
Just to flag, this was initially prompted by @mrice32 feedback here: https://github.com/across-protocol/relayer-v2/pull/1122#pullrequestreview-1791533176

**Review by mrice32 — APPROVED** [review]
LGTM!

File: scripts/spokepool.ts [hunk 1]
```diff
@@ -342,39 +342,39 @@ async function run(argv: string[]): Promise<boolean> {
...
+  return result ? NODE_SUCCESS : NODE_APP_ERR;
 }
 
 if (require.main === module) {
   run(process.argv.slice(2))
-    .then(async () => {
-      // eslint-disable-next-line no-process-exit
-      process.exit(0);
-    })
+    .then(async (result) => (process.exitCode = result))
```
Comment (mrice32) [inline scripts/spokepool.ts:375]: Returning the exit code does seem simpler, especially if we set the exitCode in exactly one function (which you are doing here).

Would make a minor nit suggestion to just expand this into {} and not use the assignment as an expression.

Comment (james-a-morris) [reply scripts/spokepool.ts:375]: +1 the resultant of the `=` operator doesn't feel very intuitive to read.

File: scripts/hubpool.ts [hunk 2]
```diff
@@ -246,34 +243,33 @@ async function run(argv: string[]): Promise<number> {
 
   config();
 
+  const cmd = argv[0];
```
Comment (james-a-morris) [inline scripts/hubpool.ts:246]: Nice break of `argv[0]` into a variable for later use

**Review by james-a-morris — APPROVED** [review]

File: scripts/spokepool.ts (hunk 1 above)
Comment (pxrl) [reply scripts/spokepool.ts]: Addressed here: https://github.com/across-protocol/relayer-v2/pull/1130/commits/d4493f53d80f2cdd2dd25afeb20fbee38e6573d2

**Pull request merged**